src/lib/aave_v3_a_token.svelte

```diff
@@ -0,0 +1,87 @@
+<script>
+    import { invoke } from "@tauri-apps/api/tauri";
+    export let token = "";
+    export let amount = "";
+    export let asset = "";
+    export let use_as_collateral = false;
+    export let category_id = 0;
+
+    /// async functions communicating with protos
+    
+    async function scheduleDeposit() {
+        const result = await invoke("DepositToAave", { token, amount });
```

Comment:
So in my head, the only thing calling `invoke()` would be the function that takes all form data and stuffs it into a `ScheduleRequest`:

```proto
message ScheduleRequest {
    // The ID (currently simply an Ethereum address) of the target Cellar
    string cellar_id = 1;
    // The block height at which to schedule the contract call
    uint64 block_height = 2;
    // The data from which the desired contract function will be encoded
    oneof call_data {
        AaveV2Stablecoin aave_v2_stablecoin = 3;
        CellarV1 cellar_v1 = 4;
        CellarV2 cellar_v2 = 5;
        CellarV2_2 cellar_v2_2 = 6;
        CellarV2_5 cellar_v2_5 = 7;
    }
    // The ID of the chain on which the target Cellar resides
    uint64 chain_id = 8;
    // The unix timestamp deadline for the contract call to be executed
    uint64 deadline = 9;
}
```

These are written as if they pass just the adaptor call to the backend instead of wrapping it into another form that submits the `ScheduleRequest`. Were you picturing a separate submission to the backend including the block height, cellar ID, chain ID, etc.?

Reply:
What I mean is a top level form that has the scheduling/chain info. This form has a field for the call_data type, and that gets populated without another svelte component of a form matching the fields for that proto, etc. and then if you do `CallOnAdaptor` as the function, each adaptor has component forms. Then one giant js object with all of the fields gets passed down to the backend and parsed as a proto. Doest this make sense or is there a better way?

Reply:
No I think what we want to do is pass the specific fields for the specific adapter in the call we want to make into the rust backend and then in the rust backend construct  a cellar_v2 message  and the create the Schedule request. https://github.com/PeggyJV/steward/blob/main/proto/steward/v4/cellar_v2.proto#L49-L83